src/ratelimit.rs

```diff
@@ -67,18 +67,18 @@ impl RLEntry {
     /// duration is the duration in milliseconds for which the hits are allowed
     fn hit(&mut self, size: u32, duration: u32) -> bool {
         let diff = Instant::now().duration_since(self.epoch);
-        let mut now = u32::try_from(diff.as_millis()).unwrap();
+        let now = u32::try_from(diff.as_millis()).unwrap();
 
         let index = usize::try_from(self.index).unwrap();
         {
             if index == self.timestamps.len() {
                 let max = usize::try_from(size).unwrap();
                 let increment = cmp::min(BLOCK_SIZE, max - index);
-                self.timestamps.extend(vec![0; increment]);
+                self.timestamps.resize(self.timestamps.len() + increment, 0);
             }
         }
 
-        now = self.rebase(now);
+        let now = self.rebase(now);
```

Comment:
Like that, an exception is : when you want this value observable for debugging with `dbg!()` or debugger.